### PR TITLE
[WebProfilerBundle] Improve AJAX toolbar panel

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Before

image

image

After

image

image

### CODE REVIEW

**Comment by javiereguiluz** [comment]
A quick question: when there are no Ajax requests, the entire Ajax panel is hidden. Do you really see it when you have 0 requests?

**Comment by ro0NL** [comment]
Yes on hover.. and it's ugly ;-)

**Comment by ro0NL** [comment]
Hmz interesting. I also noticed `span.sf-toolbar-ajax-requests` implicitly inherits styling from `table.sf-toolbar-ajax-requests` (hence the class rename).

**Comment by iltar** [comment]
Hmm maybe we need a new tag for this "improvement"?

**Comment by ro0NL** [comment]
Updated. Havent tracked back where this got broken, so im not sure how it was done before.

**Comment by ro0NL** [comment]
It seems to behave the same as latest 3.2 now, only difference i spotted 3.2 handles failures a bit different;

image

There's no red block in 3.3

**Comment by fabpot** [comment]
@javiereguiluz What do we do with this one? Is it a bug that should be fixed in 3.2 as well? Do we have a regression in current master (or even several as noted in the latest @ro0NL's comment)?

**Comment by fabpot** [comment]
@javiereguiluz Can you have a look at this one?

**Comment by ro0NL** [comment]
@fabpot  @javiereguiluz regression is caused by #20197 (see https://github.com/symfony/symfony/pull/20197/files#diff-de00cc2fe56e59a48a4d26e7a0dccca3L133)

**Comment by ro0NL** [comment]
Investigated it a bit further.. 

The display:none not being applied initially behaves really weird; after a few refreshes it works.

```js
var requestCounter = document.querySelector('.sf-toolbar-ajax-requests');
if (!requestCounter) {
   return;
}
```

seems to false-positive sometimes.

The second issue with the red status block not working is because the JS promise gets rejected due CORS policy (same as in 3.2), however we dont reach `finishAjaxRequest` in that case (for 404 responses etc. it works as expected).

**Comment by ro0NL** [comment]
@fabpot @javiereguiluz  we're all good :+1: 

Part of the regression was caused by a missing else case, which actually happened before (https://github.com/symfony/symfony/pull/20197/files#diff-de00cc2fe56e59a48a4d26e7a0dccca3L138)

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig [hunk 1]
```diff
@@ -498,7 +498,6 @@
     Sfjs.addEventListener(window, 'load', function() {
         Sfjs.createTabs();
         Sfjs.createToggles();
-        Sfjs.renderAjaxRequests();
```
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig]: This file is included in toolbar_js first.. so for being super fast this actually wins from the XHR response :))

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Comment by patrick-mcdougle** [comment]
@ro0NL Can you explain to me what I broke so I know where I need to improve in the future? Was there a regression/problem caused by me in #20197 ?

**Comment by ro0NL** [comment]
@patrick-mcdougle the following `else` was missing; https://github.com/symfony/symfony/pull/20197/files#diff-de00cc2fe56e59a48a4d26e7a0dccca3L137

https://github.com/symfony/symfony/pull/20197/files#diff-de00cc2fe56e59a48a4d26e7a0dccca3R488 caused a race condition between window & xhr load.

https://github.com/symfony/symfony/pull/20197/files#diff-de00cc2fe56e59a48a4d26e7a0dccca3R259 didnt finish ajax requests in case of errors (promise failure).

**Comment by patrick-mcdougle** [comment]
@ro0NL Thanks! I understand the race condition and the promise failure, but the `else` is right here: https://github.com/symfony/symfony/pull/20197/files#diff-de00cc2fe56e59a48a4d26e7a0dccca3R199

Is there another missing else?

**Comment by ro0NL** [comment]
The if from L193 had none ;-)

**Pull request closed without merging**